pkg/build/gobuild.go

```diff
@@ -985,6 +989,14 @@ func (g *gobuild) buildAll(ctx context.Context, ref string, baseIndex v1.ImageIn
 			if err != nil {
 				return err
 			}
+
+			// Decorate the image with the ref of the index, and the matching
+			// platform's digest.
+			baseImage = mutate.Annotations(baseImage, map[string]string{
+				specsv1.AnnotationBaseImageDigest: desc.Digest.String(),
+				specsv1.AnnotationBaseImageName:   baseRef.Name(),
```

Comment:
So this feels weird. The platform-specific image's annotations will point to the platform-specific base image's digest (good), and containing the _index's_ ref?

I think I would have expected this to be the platform-specific image's ref(-by-digest, like we get when we base an image on an image-by-digest), rather than the two annotations effectively pointing to two different things.

Are you doing this so the platform-specific image's SBOM can point up to the containing index? That seems like information we can inject into the SBOM without having to pass it through these annotations.

Reply:
I agree it is somewhat weird.  Beyond the digest being accurate (and the ref present) there's no real agenda here.

Preserving `name` felt like a way to leave sufficient breadcrumbs for the use case @jonjohnsonjr had in mind (monitoring the actual `FROM` for updates) while `digest` provides the most accurate variant of the image we actually used as our basis.

I could make these `repo@digest` vis `baseRef.Context().Digest(desc.Digest.String())` or just the `repo` via `baseRef.Context().String()`.  Ultimately as long as `repo` is there, the additional information feels somewhat extraneous, which is why I biased towards enabling @jonjohnsonjr use case (albeit not via simple `HEAD`)

Reply:
I see. You just need `image.name` to include the repo, Jon wants a mutable ref so he can HEAD to see if a rebase is needed, and I (seem to) want repo@digest so that the `image.digest` and `image.name` are both pointing to the same thing.

Thinking this through more I think I'm coming around to Jon's standpoint, even though it is weird. If a multiarch image changes but a platform-specific image within it doesn't*, then the rebase detection logic will have to do more than a HEAD to tell (maybe a GET and a HEAD, or two HEADs).

*this is somewhat relevant to how multiarch base images are updated on Dockerhub -- when `alpine:latest` is updated e.g., when `3.17` is released, a build farm kicks of images for each arch, and the index is updated in place as builds complete. So you might see `alpine:latest` update to point to `alpine[linux/amd64]:3.16` and `alpine[linux/arm64]:3.17`, then update again so they both include `:3.17`. If there are N platforms in the manifest list, it may get updated N times. If each of those updates kick off a rebase flow that is a noop (N-1)/N times, that's kinda wasteful. But I'm not really sure how else to address it and retain the workflow we wanted.

I didn't realize this was how things worked when we talked through the annotations forever ago. 😢

Reply:
I'm happy to resolve/unblock, but maybe add a comment to save me a 🤨 at this in a few months.

Reply:
> If each of those updates kick off a rebase flow that is a noop (N-1)/N times, that's kinda wasteful

Right, this was sort of my (poorly articulated) point.  The initial check may be more expensive than a HEAD, but as-is you fundamentally lack the information to judge whether it is a NOP or not without doing the work (we don't know the original digest of the base *image*, only that the index has changed).  For rebasing, the work to be redone is pretty light, but rebasing is kinda gross and if someone were actually using this to drive rebuilds, they would be in for a pretty bad time.